pkg/execution/state/redis_state/queue_processor.go

```diff
@@ -970,6 +1030,12 @@ func (q *queue) process(ctx context.Context, p QueuePartition, qi QueueItem, s *
 			Tags:    map[string]any{"status": "completed"},
 		})
 
+		if isContinuation {
+			// Add the partition to be continued again.  Note that if we've already
+			// continued beyond the limit this is a noop.
```

Comment:
If this is a noop in some cases, we may consider whether to use a more complicated locking arrangement that allows calling `addContinue` without _always_ taking a lock that might be unnecessary.